pkg/tsdb/sqleng/sql_engine.go

```diff
@@ -93,32 +124,32 @@ func (e *dataPlugin) transformQueryError(err error) error {
 	return e.queryResultTransformer.TransformQueryError(err)
 }
 
-// NewDataPlugin returns a new plugins.DataPlugin
-//nolint: staticcheck // plugins.DataPlugin deprecated
-func NewDataPlugin(config DataPluginConfiguration, queryResultTransformer SqlQueryResultTransformer,
-	macroEngine SQLMacroEngine, log log.Logger) (plugins.DataPlugin, error) {
-	plugin := dataPlugin{
+func NewQueryDataHandler(config DataPluginConfiguration, queryResultTransformer SqlQueryResultTransformer,
```

Comment:
A potential nice addition: If `*DataSourceHandler` implements the [`Dispose` interface](https://github.com/grafana/grafana-plugin-sdk-go/blob/843e160c3b78b11a1db78e3a477f3fc3ec20e9c9/backend/instancemgmt/instance_manager.go#L19-L21) it could call `Close()` on the "old" Xorm engine. Could be considered later though since I'm not sure it could have side effects.